src/lib.rs

```diff
@@ -256,8 +256,16 @@ impl CreateOptions {
 		Self::default()
 	}
 
+	/// Create a new [`MemFile`]` with the given options.
+	///
+	/// This is a shorthand for [`MemFile::create`].
+	/// See that function for more details.
+	pub fn create(&self, name: &str) -> std::io::Result<MemFile> {
+		MemFile::create(name, *self)
+	}
+
```

Comment:
Generally, builder use a `build()` method name.

This is unsound cause it's doesn't take self by value.

Reply:
How does that make it unsound? You can resuse the same options many times without issue.

I really prefer the name `create` by the way, which is in line with [`std::fs::OpenOptions`](https://doc.rust-lang.org/stable/std/fs/struct.OpenOptions.html#method.create).

Reply:
but you could reuse the same option many with self too cause it's implement copy, it's unsound cause now create take OptionsCreate by value, since now OptionsCreate have a create method, I think they both should either take by value the Option or both take by ref.

Reply:
It may be a little inconsistent, but since this is a member function you won't notice any difference in using it. Specifically, there is no unsoundness, as we're not violating any safety requirements.